packages/loader/driver-utils/src/runWithRetry.ts

```diff
@@ -129,3 +131,19 @@ export async function runWithRetry<T>(
 	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 	return result!;
 }
+
+const MaxReconnectDelayInMsWhenEndpointIsReachable = 30000;
+const MaxReconnectDelayInMsWhenEndpointIsNotReachable = 8000;
```

Comment:
There is potential danger here as well. In case backend is misconfigured and does not return proper CORS headers then it will be treated as not 'reached' while in fact request has been handled by backend and response has been sent. This is super rare but can happen. I think 8 sec is safe only when we have strong indicator that user is offline.

Reply:
I check the code. In case there is some error response with no status code, we will call throwOdspNetworkError(), here: https://github.com/microsoft/FluidFramework/blob/e22acc32afad5422b046cf748e87b6bda58398c2/packages/drivers/odsp-driver/src/odspUtils.ts#L122
as response.ok would be false. In case of throwOdspNetworkError, we always set endpointReached as true irrespective of status code.

Reply:
I think the biggest issue here - we do not have strong signal that we are offline. So, we assume that any case where fetch() failed is offline/not-reached, and any case where any response was returned is service reachable.
If we have cases where service responded, but we did not get a response at all, then we can't detect that - this is same as offline